### PR TITLE
Jingxuan/map

### PR DESCRIPTION
**Pull request by Taosit**
- Implemented search with autocompletion for locations
- Displayed location the user selects on map
- Implemented endpoints that make requests to google map api
- Displayed the correct location on map on listing details page

### CODE REVIEW

File: app/page.tsx [hunk 1]
```diff
@@ -11,6 +11,8 @@ interface HomeProps {
   searchParams: IListingsParams;
 }
 
+export const dynamic = "force-dynamic";
```
Comment (Shanni) [inline app/page.tsx:14]: looks fancy

File: app/components/modals/SearchModal.tsx [hunk 2]
```diff
@@ -129,10 +129,10 @@ const SearchModal = () => {
       />
       <CountrySelect
         value={location}
-        onChange={(value) => setLocation(value as CountrySelectValue)}
+        onCountryChange={(value) => setLocation(value)}
       />
       <hr />
-      <Map center={location?.latlng} />
+      <Map coordinates={[0, 0]} />
```
Comment (Shanni) [inline app/components/modals/SearchModal.tsx:135]: Curious, why [0,0]

File: app/hooks/useLocationSearch.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  const selectLocation = async (place: Place) => {
+    setImmediateValue(place.name);
+    const placeDetails = await getPlaceDetails(place.id);
+    onChange(placeDetails);
+    setSearchString(place.name);
+    setShouldShowPlaces(false);
+  };
+
+  const handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === "Escape") {
```
Comment (Shanni) [inline app/hooks/useLocationSearch.ts:53]: wow looks fancy, seems to be a hidden feature

File: app/hooks/useLocationSearch.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+      setShouldShowPlaces(true);
+    }
+  };
+
+  const handleClearSearch = () => {
+    setImmediateValue("");
+    onChange({
+      address: "",
+      coordinates: {
+        lat: 51,
```
Comment (Shanni) [inline app/hooks/useLocationSearch.ts:65]: how about "The latitude of Vancouver, BC, Canada is 49.246292, and the longitude is -123.116226" :DDDDD

**Review by Shanni — COMMENTED** [review]
Thanks!! All looks amazing!

small thing, for structure, placeDetails and placeAutoComplete could make to one dir, e.g. naming `locations`, I would try to group them based on features, and both are calling Google map, so you can have request like `${baseUrl}/api/path`

**Review by Shanni — APPROVED** [review]
Thanks!! All looks amazing!

small thing, for structure, placeDetails and placeAutoComplete could make to one dir, e.g. naming `locations`, I would try to group them based on features, and both are calling Google map, so you can have request like `${baseUrl}/api/path`

File: app/components/modals/SearchModal.tsx (hunk 2 above)
Comment (Taosit) [reply app/components/modals/SearchModal.tsx:135]: You're right, I'm going to set the default location to Vancouver

File: app/hooks/useLocationSearch.ts (hunk 3 above)
Comment (Taosit) [reply app/hooks/useLocationSearch.ts:53]: I'm thinking I probably need to improve keyboard navigation more. I'll do that under this pr as well before I merge

**Pull request merged**